pkg/apis/componentconfig/types.go

```diff
@@ -392,7 +392,7 @@ type KubeControllerManagerConfiguration struct {
 	// horizontalPodAutoscalerTolerance is the tolerance for when
 	// resource usage suggests upscaling/downscaling
 	HorizontalPodAutoscalerTolerance float64
-	// deploymentControllerSyncPeriod is the period for syncing the deployments.
+	// DEPRECATED: deploymentControllerSyncPeriod is the period for syncing the deployments.
```

Comment:
Can this not be removed completely?

Reply:
Not yet. It is still referenced from cmd/kube-controller-manager/app/options/options.go

In theory, this can be completely removed. But doing that may break users who are already using this flag.
KCM may refuse to start when encountering an unrecognizable flag.
That is the only reason we are supposed to leave it here for a few months.